src/libs/actions/Report.js

```diff
@@ -380,7 +380,13 @@ function fetchChatReports() {
  * @param {Number} reportID
  */
 function fetchActions(reportID) {
-    API.Report_GetHistory({reportID})
+    API.Report_GetHistory({
+        reportID,
+
+        // Get the most recent action we have and provide this as an offset
+        // to prevent refetching large sets of data we already have
+        offset: reportMaxSequenceNumbers[reportID] || 0,
```

Comment:
Another thing to watch out for is any situation where a reportMaxSequenceNumber for a report should be reset e.g. when evicting the report from storage.